src/gsi/gsi_fedOper.F90

```diff
@@ -0,0 +1,185 @@
+  end function nodeMold
+
+  subroutine setup_(self, lunin, mype, is, nobs, init_pass, last_pass)
+    use fed_setup, only: setup
+    use kinds, only: i_kind
+    use gsi_obOper, only: len_obstype
+    use gsi_obOper, only: len_isis
+
+    use m_rhs  , only: awork => rhs_awork
+    use m_rhs  , only: bwork => rhs_bwork
+    use m_rhs  , only: iwork => i_fed
+
+    use obsmod  , only: write_diag
+    use jfunc   , only: jiter
+
+    use mpeu_util, only: die
+
+    use obsmod, only: dirname, ianldate
+
+    implicit none
+    class(fedOper ), intent(inout):: self
+    integer(i_kind), intent(in):: lunin
+    integer(i_kind), intent(in):: mype
+    integer(i_kind), intent(in):: is
+    integer(i_kind), intent(in):: nobs
+    logical        , intent(in):: init_pass     ! supporting multi-pass setup()
+    logical        , intent(in):: last_pass     ! with incremental backgrounds.
+
+    !----------------------------------------
+    character(len=*),parameter:: myname_=myname//"::setup_"
+
+    character(len=len_obstype):: obstype
+    character(len=len_isis   ):: isis
+    integer(i_kind):: nreal,nchanl,ier,nele
+    logical:: diagsave
+    integer(i_kind):: lu_diag
+    character(128):: diag_file
+    character(80):: string
+
+    if(nobs == 0) then
+
+       if( (mype == 0) .and. init_pass ) then
+          write(string,600) jiter
+600       format('fed_',i2.2)
+          diag_file=trim(dirname) // trim(string)
+          write(6,*) 'write ianldate to ', diag_file
+          open(newunit=lu_diag,file=trim(diag_file),form='unformatted',status='unknown',position='rewind')
+          write(lu_diag) ianldate
+          close(lu_diag)
+       endif
+
+       return
+
+    endif
+
+    read(lunin,iostat=ier) obstype,isis,nreal,nchanl
+    if(ier/=0) call die(myname_,'read(obstype,...), iostat =',ier)
+    nele = nreal+nchanl
+
+    diagsave  = write_diag(jiter) .and. diag_fed
+
+    call setup(self%obsLL(:), self%odiagLL(:), &
+        lunin,mype,bwork,awork(:,iwork),nele,nobs,is,diagsave,init_pass)
+
+  end subroutine setup_
+
+  subroutine intjo1_(self, ibin, rval,sval, qpred,sbias)
+!    use intfedmod, only: intjo => intfed
+    use gsi_bundlemod  , only: gsi_bundle
+    use bias_predictors, only: predictors
+    use m_obsNode , only: obsNode
+    use m_obsLList, only: obsLList_headNode
+    use kinds     , only: i_kind, r_quad
+    implicit none
+    class(fedOper  ),intent(in   ):: self
+    integer(i_kind ),intent(in   ):: ibin
+    type(gsi_bundle),intent(inout):: rval   ! (ibin)
+    type(gsi_bundle),intent(in   ):: sval   ! (ibin)
+    real(r_quad    ),target,dimension(:),intent(inout):: qpred  ! (ibin)
+    type(predictors),target,             intent(in   ):: sbias
+
+    !----------------------------------------
+    character(len=*),parameter:: myname_=myname//"::intjo1_"
+    class(obsNode),pointer:: headNode
+
+!    headNode => obsLList_headNode(self%obsLL(ibin))
+!    call intjo(headNode, rval,sval)
+!    headNode => null()
```

Comment:
Do we need line 151-153? If not, suggest removing them

Reply:
We don't currently need this code.  I agree that it is good to remove these lines.